Fix Quick Start Guide Link in README.md

LUC18fknU7P opened the pull request:
The previous link https://docs.helm.sh/using_helm/#quickstart-guide was redirecting into https://helm.sh/ja/docs/intro/ which is a Japanese version of the docs. I've fixed the link so it goes into the English version instead.

<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Make sure to read the Contributing Guide before submitting your PR: https://github.com/helm/helm/blob/master/CONTRIBUTING.md
2. If this PR closes another issue, add 'closes #<issue number>' somewhere in the PR summary. GitHub will automatically close that issue when this PR gets merged. Alternatively, adding 'refs #<issue number>' will not close the issue, but help provide the reviewer more context.-->

**What this PR does / why we need it**:
This fixes the **Quick Start Guide** link in the **[Install](https://github.com/helm/helm/blob/master/README.md#Install)** section of [README.md](https://github.com/helm/helm/blob/master/README.md)
**Special notes for your reviewer**:
None
**If applicable**:
- [x] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

hickeyma reviewed (changes requested):
Good catch. thanks for finding and fixing this @tero-dev . It looks good except that you need to sign the dco. See "What If I Forget To Sign-off On A Commit" section in [Helm DCO](https://helm.sh/blog/helm-dco/) doc for details on how to sign it.

LUC18fknU7P committed:
Fix Quick Start Guide Link in README.md

The previous link https://docs.helm.sh/using_helm/#quickstart-guide was redirecting into https://helm.sh/ja/docs/intro/ which is a Japanese version of the docs. I've fixed the link so it goes into the English version instead.

Signed-off-by: Tero <31454692+tero-dev@users.noreply.github.com>

tero-dev commented:
> Good catch. thanks for finding and fixing this @tero-dev . It looks good except that you need to sign the dco. See "What If I Forget To Sign-off On A Commit" section in [Helm DCO](https://helm.sh/blog/helm-dco/) doc for details on how to sign it.

Sorry that it took a while, I couldn't get back immediately so i had to push this off for a day but now the commit has a DCO signature. Should be all good now.

hickeyma reviewed (approved):
LGTM, thanks @tero-dev
